src/core/database/database.service.ts

```diff
@@ -604,117 +603,126 @@ export class DatabaseService {
-            id: "${session.userId}",
-            ${aclEditProp}: true
-          })
-        CREATE
-          (item:${upperFirst(baseNodeLabel)} {
-            active: true,
-            createdAt: datetime(),
-            id: "${input.id}",
-            owningOrgId: "${session.owningOrgId}"
-          })<-[:toNode]-(acl:ACL    
-            ${aclString}
-          )-[:member]->(requestingUser)
-        RETURN item
-      `;
     try {
       await this.db
         .query()
-        .raw(query, {
-          requestingUserId: session.userId,
-          token: session.token,
-          id: input.id,
-        })
+        .match([
+          node('token', 'Token', {
+            active: true,
+            value: session.token,
+          }),
+          relation('in', '', 'token', {
+            active: true,
+          }),
+          node('requestingUser', 'User', {
+            active: true,
+            id: session.userId,
+            ...(aclEditProp ? { [aclEditProp]: true } : {}),
+          }),
+        ])
+        .create([
+          node('item', upperFirst(baseNodeLabel), {
+            active: true,
+            createdAt: DateTime.local(),
+            id: input.id,
+            owningOrgId: session.owningOrgId,
+          }),
+          relation('in', '', 'toNode'),
+          node('acl', 'ACL', acls),
+          relation('out', '', 'member'),
+          node('requestingUser'),
+        ])
         .run();
     } catch (e) {
-      const ACLQuery = `MATCH
-      (token:Token {
-        active: true,
-        value: "${session.token}"
-      })
-      <-[:token {active: true}]-
-      (requestingUser:User {
-        active: true,
-        id: "${session.userId}"
-      })
-      RETURN requestingUser.${aclEditProp}
-      `;
+      // If there is no aclEditProp, then this is not an access-related issue and we can move forward with throwing
+      if (!aclEditProp) {
+        throw e;
+      }
 
-      const result = ((await this.db
+      // Retrieve the user's record of the aclEditProp, if it exists
+      const aclResult = await this.db
         .query()
-        .raw(ACLQuery, {})
-        .run()) as unknown) as { requestingUser: boolean } | null;
-      if (!result || !result.requestingUser) {
+        .match([
+          node('token', 'Token', {
+            active: true,
+            value: session.token,
+          }),
+          relation('in', '', 'token'),
+          node('requestingUser', 'User', {
+            active: true,
+            id: session.userId,
+          }),
+        ])
+        .return({
+          requestingUser: [{ aclEditProp: 'editProp' }],
+        })
+        .first();
+
+      // If the user doesn't have permission to perform the create action...
+      if (!aclResult || !aclResult.editProp) {
         throw new ForbiddenError(`${aclEditProp} missing or false`);
       }
-      this.logger.error(`${e} create Node error`);
+
+      this.logger.error(`DatabaseService create error`, {
```

Comment:
The loggers are always scoped to a subject/service/namespace so prefixing isn't necessary.